### PR TITLE
Spaceships example: fix Visual Interpolation bits

### PR DESCRIPTION
**Pull request by RJ**
Clarifies how the visual interp is set up, and fixes the syncplugin issue as described in the lightyear book VisualInterpolation chapter

This removes the stutter from the text labels that follow the players, since they are positioned with `Transform`, not `Position`

Also filters collisions between players and their own bullets, because it was possible to collide with your own bullets while accelerating.

### CODE REVIEW

**Comment by RJ** [comment]
`cargo test` passes for me locally, not sure why it's failing here

File: examples/spaceships/src/renderer.rs [hunk 1]
```diff
@@ -63,10 +63,48 @@ impl Plugin for SpaceshipsRendererPlugin {
...
+// (Note also that we've configured avian's SyncPlugin to run in PostUpdate)
+fn add_visual_interpolation_components<T: Component>(
+    trigger: Trigger<OnAdd, T>,
+    q: Query<Entity, (With<T>, Without<Wall>, Without<Confirmed>)>,
+    mut commands: Commands,
+) {
+    if !q.contains(trigger.entity()) {
+        return;
     }
+    info!("Adding visual interp component to {:?}", trigger.entity());
```
Comment (cBournhonesque) [inline examples/spaceships/src/renderer.rs:101]: We might want this log as debug

**Review by cBournhonesque — APPROVED** [review]
Thanks this looks great to me!

One of the unit tests is a bit flimsy

File: examples/spaceships/src/renderer.rs (hunk 1 above)
Comment (RJ) [reply examples/spaceships/src/renderer.rs]: agreed , changed

**Pull request merged**